### PR TITLE
[INLONG-4397][Agent] Supports collect of full data for file type

### LINKED ISSUE #4397: [Feature][Agent] Supports collect of full data for file type
### Description

Supports collect of full data for file types Support

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Greedyu**
Supports collect of full data for file type

Fixes #4397
image

### CODE REVIEW

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/JobConstants.java [hunk 1]
```diff
@@ -48,6 +48,7 @@ public class JobConstants extends CommonConstants {
     public static final String JOB_FILE_TIME_OFFSET = "job.fileJob.timeOffset";
     public static final String JOB_FILE_MAX_WAIT = "job.fileJob.file.max.wait";
     public static final String JOB_CYCLE_UNIT = "job.fileJob.cycleUnit";
+    public static final String JOB_FILE_SYNC_TYPE = "job.fileJob.syncType";
```
Comment (dockerzhang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/JobConstants.java:51]: `collect` is better than `sync`

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/SyncTypeConstants.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.constant;
+
+/**
+ * Data source synchronization type
+ */
+public class SyncTypeConstants {
```
Comment (dockerzhang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/SyncTypeConstants.java:23]: SyncTypeConstants
->
FileCollectType

**Comment by dockerzhang** [comment]
@Greedyu I think it's better to add a configuration to limit the file number.

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/JobConstants.java (hunk 1 above)
Comment (Greedyu) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/JobConstants.java]: resolve

**Comment by Greedyu** [comment]
> @Greedyu I think it's better to add a configuration to limit the file number.
The judgment of the file limit is logically processed in the TextFileSource.split() method to read the file
wecom-temp-57a16e6f48e9d5956ff5ff55c3de70ba

**Comment by dockerzhang** [comment]
@EMsnap @pocozh PTAL, thanks.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java [hunk 3]
```diff
@@ -144,6 +146,21 @@ public void testOneJobOnly() throws Exception {
...
+    public void testOneJobFullPath() throws Exception {
+        String jsonString = TestUtils.getTestTriggerProfile();
+        TriggerProfile triggerProfile = TriggerProfile.parseJsonStr(jsonString);
+        triggerProfile.set(JOB_DIR_FILTER_PATTERN,
+                Paths.get(getClass().getClassLoader().getResource("test").toURI()).toString());
+        triggerProfile.set(JOB_FILE_MAX_WAIT, "-1");
+        triggerProfile.set(JOB_FILE_COLLECT_TYPE, FileCollectType.FULL);
+        TriggerManager triggerManager = agent.getManager().getTriggerManager();
+        triggerManager.addTrigger(triggerProfile);
+        while (true) {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java:159]: what's the purpose of this code

Comment (Greedyu) [reply inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java]: In full mode, you can add files to the target after running to test whether the increment is normal

**Review by pocozh — APPROVED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java (hunk 3 above)
Comment (EMsnap) [reply inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java]: runing a dead loop is not proper

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java [hunk 4]
```diff
@@ -144,6 +146,19 @@ public void testOneJobOnly() throws Exception {
         Assert.assertTrue(checkOnlyOneJob());
     }
 
+    // @Test
+    public void testOneJobFullPath() throws Exception {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java:150]: the test cannot test full mode

Comment (Greedyu) [reply inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java:162]: The trigger will be submitted in the method, and the full file will be read

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java (hunk 3 above)
Comment (Greedyu) [reply inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/TestFileAgent.java]: It has been modified to add files to the code to test the addition of full mode

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/FileCollectType.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.constant;
+
+/**
+ * Data source synchronization type
```
Comment (healchow) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/FileCollectType.java:21]: It is suggested to comment it with `Collect type of the File data`.

**Review by EMsnap — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/TextFileSource.java [hunk 6]
```diff
@@ -75,6 +75,7 @@ public List<Reader> split(JobProfile jobConf) {
         for (File file : allFiles) {
             int seekPosition = jobConf.getInt(file.getAbsolutePath() + POSITION_SUFFIX, 0);
             LOGGER.info("read from history position {} with job profile {}", seekPosition, jobConf.getInstanceId());
+            LOGGER.info("file absolute path: {}", file.getAbsolutePath());
```
[the remark this answers is not in the document]
Comment (Greedyu) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/TextFileSource.java]: resolve

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/FileCollectType.java (hunk 5 above)
Comment (Greedyu) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constant/FileCollectType.java]: resolve

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**